rbac/management/audit_log/model.py

```diff
@@ -58,3 +62,63 @@ class AuditLog(TenantAwareModel):
     resource_id = models.IntegerField(null=True)
     action = models.CharField(max_length=32, choices=ACTION_CHOICES)
     tenant = models.ForeignKey(Tenant, on_delete=models.SET_NULL, null=True)
+
+    def get_resource_item(self, r_type, request, *args, **kwargs):
+        """Find different values of various fields."""
+        if r_type == AuditLog.ROLE:
+            role_items = []
+            if request.data != {}:
+                role_object = get_object_or_404(Role, name=request.data["name"])
+                role_object_id = role_object.id
+                role_object_name = "role: " + role_object.name
+                role_items.append(role_object_id)
+                role_items.append(role_object_name)
+                return role_items
+            else:
+                role_object = kwargs["kwargs"]
+                role_object_id = role_object.id
+                role_object_name = "role: " + role_object.name
+                role_items.append(role_object_id)
+                role_items.append(role_object_name)
+                return role_items
+
+        elif r_type == AuditLog.GROUP:
+            if request._data is not None:
+                group_object = get_object_or_404(Group, name=request.data["name"])
+            else:
+                group_uuid = kwargs["kwargs"]["uuid"]
+                group_object = get_object_or_404(Group, uuid=group_uuid)
+            return group_object.id
+
+        elif r_type == AuditLog.PERMISSION:
+            # TODO: finding the id for the permission
+            return None
+
+        elif r_type == "principal":
+            current_user = management.utils.get_principal_from_request(request)
+            principal_object = get_object_or_404(Principal, username=current_user.username)
+            list_principal_object = [principal_object.id, principal_object.username]
+            return list_principal_object
+
+        elif r_type == "tenant":
+            tenant_object = get_object_or_404(Tenant, org_id=request._user.org_id)
+            return tenant_object.id
+
+        else:
+            return ValueError("This resource type does not exist or is not role/group/tenant/permission/principal")
+
+    def log_create(self, request, resource):
+        """Audit Log when group or user is created."""
+        principal_items = self.get_resource_item("principal", request)
+        self.principal_id = principal_items[0]
+        self.principal_username = principal_items[1]
+
+        create_resource_items = self.get_resource_item(resource, request)
+
+        self.resource_type = resource
+        self.resource_id = create_resource_items[0]
```

Comment:
sometimes is the return value from `get_resource_item()` list with 2 items and sometimes something different (e.g. group id), this will raise the `TypeError` on this line